vars/handlePipelineStepErrors.groovy

```diff
@@ -64,6 +64,8 @@ void call(Map parameters = [:], body) {
         .addIfEmpty('stepNameDoc' , parameters.stepName)
         .use()
 
+    // load 'unstableSteps' here as loading it later will result in list transforming to JSONArray (only in piperExecuteBin)
```

Comment:
The comment does not tell my anything. Would be worth IMO to explain the details.

Reply:
Better now?

Reply:
Yes. That behaviour looks confusing. Out of curiosity: Do we understand why it behaves differently when inside a body closure.

Reply:
It doesn't behave differently in `body`. But when `piperExecuteBin` is used, we write the CPE to disk using `JSON`. This changes the type from groovy-based to json-based.